src/ImageSharp/Formats/Tiff/TiffDecoderCore.cs

```diff
@@ -278,71 +302,57 @@ private ImageFrame<TPixel> DecodeFrame<TPixel>(ExifProfile tags, CancellationTok
                 stripByteCounts,
                 cancellationToken);
         }
-
-        return frame;
-    }
-
-    private IMemoryOwner<ulong> ConvertNumbers(Array array, out Span<ulong> span)
-    {
-        if (array is Number[] numbers)
-        {
-            IMemoryOwner<ulong> memory = this.memoryAllocator.Allocate<ulong>(numbers.Length);
-            span = memory.GetSpan();
-            for (int i = 0; i < numbers.Length; i++)
-            {
-                span[i] = (uint)numbers[i];
-            }
-
-            return memory;
-        }
-
-        DebugGuard.IsTrue(array is ulong[], $"Expected {nameof(UInt64)} array.");
-        span = (ulong[])array;
-        return null;
     }
 
     /// <summary>
-    /// Calculates the size (in bytes) for a pixel buffer using the determined color format.
+    /// Decodes the image data for Tiff's which arrange the pixel data in tiles.
     /// </summary>
-    /// <param name="width">The width for the desired pixel buffer.</param>
-    /// <param name="height">The height for the desired pixel buffer.</param>
-    /// <param name="plane">The index of the plane for planar image configuration (or zero for chunky).</param>
-    /// <returns>The size (in bytes) of the required pixel buffer.</returns>
-    private int CalculateStripBufferSize(int width, int height, int plane = -1)
+    /// <typeparam name="TPixel">The pixel format.</typeparam>
+    /// <param name="tags">The IFD tags.</param>
+    /// <param name="frame">The image frame to decode into.</param>
+    /// <param name="cancellationToken">The token to monitor cancellation.</param>
+    private void DecodeImageWithTiles<TPixel>(ExifProfile tags, ImageFrame<TPixel> frame, CancellationToken cancellationToken)
+        where TPixel : unmanaged, IPixel<TPixel>
     {
-        DebugGuard.MustBeLessThanOrEqualTo(plane, 3, nameof(plane));
+        Buffer2D<TPixel> pixels = frame.PixelBuffer;
+        int width = pixels.Width;
+        int height = pixels.Height;
+
+        int tileWidth = (int)tags.GetValue(ExifTag.TileWidth).Value;
+        int tileLength = (int)tags.GetValue(ExifTag.TileLength).Value;
+        int tilesAcross = (width + tileWidth - 1) / tileWidth;
+        int tilesDown = (height + tileLength - 1) / tileLength;
+
+        Array tilesOffsetsArray;
+        Array tilesByteCountsArray;
+        IExifValue tilesOffsetsExifValue = tags.GetValueInternal(ExifTag.TileOffsets);
+        IExifValue tilesByteCountsExifValue = tags.GetValueInternal(ExifTag.TileByteCounts);
+        if (tilesOffsetsExifValue is null)
+        {
+            // Note: This is against the spec, but libTiff seems to handle it this way.
```

Comment:
I've come to believe there is not a single image spec over 10 years old that hasn't been blatantly trampled on